code/game/machinery/portable_turret.dm

```diff
@@ -614,21 +616,35 @@ GLOBAL_LIST_EMPTY(turret_icons)
 		if(target(M))
 			return 1
 
-/obj/machinery/porta_turret/proc/popUp()	//pops the turret up
+/obj/machinery/porta_turret/proc/check_popUp()
+	/// Whitelist to determine what objects can be put over turrets while letting them deploy
+	var/static/list/deployment_whitelist = typecacheof(list(
+	/obj/structure/window/reinforced,
+	/obj/structure/window/basic,
+	/obj/structure/window/plasmabasic,
+	/obj/structure/window/plasmareinforced,
+	/obj/machinery/door/window,
+	))
 	if(disabled)
 		return
 	if(raising || raised)
 		return
 	if(stat & BROKEN)
 		return
+	if(deployment_override)
+		pop_up()
 	// check if anything's preventing us from raising
 	var/turf/T = get_turf(src)
 	for(var/atom/A in T)
 		if(A == src)
 			continue
 		if(A.density)
+			if(is_type_in_typecache(A, deployment_whitelist))
+				continue
 			return
```

Comment:
If you don't want to hardcode the types, you can do
```suggestion
		if(!A.CanPass(src, src.loc))
				return
```
This appears to work in my testing, with the exception that dense mobs (i.e. standing humans/bots) no longer prevent deployment.

Reply:
Unsure if that's a bit of a scuffed solution, I'll get a PRR to have a look before implementing